File: MainApp/AgentApp/Frontend/src/app/model/message.ts
import { UserInfo } from "./userInfo";
import { User } from "./user";
import { RentRequest } from "./rentRequest";

export class Message {
  id: number;
  text: string;
  sender: UserInfo;
  dateAndTime: string;

  constructor(
    text: string,
    sender: UserInfo,
    dateAndTime: string,
    id?: number
  ) {
    this.text = text;
    this.sender = sender;
    this.dateAndTime = dateAndTime;
    this.id = id;
  }
}

File: MainApp/AgentApp/Backend/src/main/java/jvn/RentACar/model/Message.java
package jvn.RentACar.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Message {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private Long mainAppId;

    @Column(nullable = false)
    private String text;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private User sender;

    @Column(nullable = false)
    private LocalDateTime dateAndTime;

    @ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    private RentRequest rentRequest;
    
}
